src/bin/addr2line.rs

```diff
@@ -1,240 +1,97 @@
-use std::env;
-use std::fs;
+use clap::{App, Arg};
 
-fn main() {
-    let args: Vec<String> = env::args().collect();
-    let mut opts = getopts::Options::new();
-    opts.optopt("e",
-                "exe",
-                "Set the input file name (default is a.out)",
-                "<executable>");
-
-    let matches = opts.parse(&args[1..]).unwrap();
+use std::path;
 
-    let file_path = matches.opt_str("e").unwrap_or("a.out".to_string());
-    let file = fs::File::open(&file_path).expect("Should open file");
-    let file = memmap::Mmap::open(&file, memmap::Protection::Read)
-        .expect("Should create a mmap for file");
-    let file = object::File::parse(unsafe { file.as_slice() });
+#[cfg_attr(rustfmt, rustfmt_skip)]
+const POSTSCRIPT: &'static str = "\
+addr2line translates addresses into file names and line numbers.
+Given an address in an executable or an offset in a section of a
+relocatable object, it uses the debugging information to figure
+out which file name and line number are associated with it.
 
-    let addrs = matches.free.iter().map(|x| parse_uint_from_hex_string(x)).collect();
+By default, the format of the output is `FILENAME:LINENO', and
+each input address generates one line of output. This behavior
+can be modified by passing the -a option, which also includes
+the address on its own line.
 
-    if file.is_little_endian() {
-        symbolicate::<gimli::LittleEndian>(&file, addrs);
-    } else {
-        symbolicate::<gimli::BigEndian>(&file, addrs);
-    }
-}
-
-fn parse_uint_from_hex_string(string: &str) -> u64 {
-    if string.len() > 2 && string.starts_with("0x") {
-        u64::from_str_radix(&string[2..], 16).expect("Failed to parse address")
-    } else {
-        u64::from_str_radix(string, 16).expect("Failed to parse address")
-    }
-}
+If the file name can not be determined, addr2line will print two
+question marks in their place. If the line number can not be
+determined, addr2line will print 0.
+";
 
-fn display_file<Endian>(unit: &Unit,
-                        header: &gimli::LineNumberProgramHeader<Endian>,
-                        row: &gimli::LineNumberRow)
-    where Endian: gimli::Endianity
-{
-    let file = row.file(header).unwrap();
-    if let Some(directory) = file.directory(header) {
-        let directory = directory.to_string_lossy();
-        if !directory.starts_with("/") {
-            if let Some(comp_dir) = unit.comp_dir() {
-                print!("{}/", comp_dir.to_string_lossy());
+fn main() {
+    let matches = App::new("addr2line")
+        .version(crate_version!())
+        .about("Convert addresses into line number/file name pairs.\n\
+                If no addresses are specified on the command line, they will be read from stdin.")
+        .arg(Arg::with_name("executable")
+            .short("e")
+            .long("exe")
+            .default_value("a.out")
+            .help("Specify the name of the executable for which addresses should be translated.")
```

Comment:
I took the help text verbatim from the `addr2line` [man page](https://sourceware.org/binutils/docs/binutils/addr2line.html) for consistency, but I'm happy to change it if you prefer path.